src/logic/deeplink/index.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  compressToEncodedURIComponent,
+  decompressFromEncodedURIComponent,
+} from 'lz-string'
+import { Serializable } from '@src/model/serializable'
+
+export interface DeepLinkPayload {
+  input: Serializable
+  reactorCode?: string
+}
+
+export function encodeToDeepLink(payload: DeepLinkPayload) {
+  return compressToEncodedURIComponent(JSON.stringify(payload))
+}
+
+export function decodeFromDeepLink(encoded: string) {
+  return JSON.parse(
+    decompressFromEncodedURIComponent(encoded),
+  ) as DeepLinkPayload
+}
```

Comment:
Note that this may throw parsing error. Currently error is ignored and just set input value as `undefined`. But I'd like to improve such case by using better parser.